packages/block-editor/src/components/global-styles/typography-panel.js

```diff
@@ -257,33 +256,16 @@ export default function TypographyPanel( {
 
 	// Check if previous font style and weight values are available in the new font family
 	useEffect( () => {
-		const { fontStyles, fontWeights, isSystemFont } =
-			getFontStylesAndWeights( fontFamilyFaces );
-		const hasFontStyle = fontStyles?.some(
-			( { value: fs } ) => fs === fontStyle
-		);
-		const hasFontWeight = fontWeights?.some(
-			( { value: fw } ) => fw === fontWeight
-		);
-
-		// Try to set nearest available font weight
-		if ( ! hasFontWeight && fontWeight ) {
-			setFontAppearance( {
-				fontStyle,
-				fontWeight: findNearestFontWeight( fontWeights, fontWeight ),
-			} );
-		}
+		const { nearestFontStyle, nearestFontWeight } =
+			findNearestStyleAndWeight( fontFamilyFaces, fontStyle, fontWeight );
 
-		// Set the same weight and style values if the font family is a system font or if both are the same
-		if ( isSystemFont || ( hasFontStyle && hasFontWeight ) ) {
+		if ( nearestFontStyle && nearestFontWeight ) {
 			setFontAppearance( {
-				fontStyle,
-				fontWeight,
+				fontStyle: nearestFontStyle,
+				fontWeight: nearestFontWeight,
 			} );
-		}
-
-		// Reset font appearance if the font family does not have the selected font style
-		if ( ! hasFontStyle ) {
+		} else {
+			// Reset font appearance if there are no available styles or weights
```

Comment:
Nit: the coding standards mention comments should end in a period, though it's not enforced by the linter, so not very important (I wouldn't worry too much about fixing ones not added in this PR).

Reply:
Thanks! I've added periods to the end of the comments added in this PR.